main.R

```diff
@@ -5,20 +5,21 @@ library(stringr)
 library(sbtools)
 library(whisker)
 
-outer_dir <- 'C:/Users/SLEUNG/Github/ds-pipelines-targets-1'
+# REQUIREMENT: working dir must be same as dir where main.R is located
+outer_dir <- getwd()
 
 print('source the following files:')
 for (dir in list('1_fetch', '2_process', '3_visualize')) {
   setwd(file.path(outer_dir, dir, 'src'))
-  src_files = list.files(pattern='\\.R$')
+  src_files <- list.files(pattern='\\.R$')
   print(paste('in ', getwd(), ': ', sep=""))
   print(src_files)
   sapply(src_files, source)
 }
```

Comment:
This will work, but I personally always prefer to avoid `getwd()` and `setwd()`. It can lead confusing bugs if your code doesn't return to the original working directory.

In R you could do this without changing your working directory with something like:

```r
src_files <- Sys.glob("*/src/*.R")
sapply(src_files, source)
# or if you want more feedback
# for (f in src_files) {
#  print(paste("sourcing", f))
#  source(f)
# }
```

I'm going to approve this PR and leave it up to you whether to make a change like this.

Reply:
Wow, that's really helpful!! I didn't like the way I was doing it and this is so much better. Thank you!